workspace/test_configurations.go

```diff
@@ -0,0 +1,230 @@
+package workspace
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+)
+
+type TestConfiguration struct {
+	// The static portion of the test Command, which will be run for every test on this track. Examples include `cargo test` or `go test`.
+	// Might be empty if there are platform-specific versions
+	Command string
+
+	// Windows-specific test command. Mostly relevant for tests wrapped by shell invocations. Falls back to `Command` if we're not running windows or this is empty.
+	WindowsCommand string
+}
+
+func (c *TestConfiguration) GetTestCommand() (string, error) {
+	var cmd string
+	if runtime.GOOS == "windows" && c.WindowsCommand != "" {
+		cmd = c.WindowsCommand
+	} else {
+		cmd = c.Command
+	}
+
+	// pre-declare these so we can conditionally initialize them
+	var exerciseConfig *ExerciseConfig
+	var err error
+
+	if strings.Contains(cmd, "{{") {
+		// only read exercise's config.json if we need it
+		exerciseConfig, err = NewExerciseConfig(".")
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if strings.Contains(cmd, "{{solution_files}}") {
+		if exerciseConfig == nil {
+			return "", fmt.Errorf("exerciseConfig not initialize before use")
+		}
+		solutionFiles, err := exerciseConfig.GetSolutionFiles()
+		if err != nil {
+			return "", err
+		}
+		cmd = strings.ReplaceAll(cmd, "{{solution_files}}", strings.Join(solutionFiles, " "))
+	}
+	if strings.Contains(cmd, "{{test_files}}") {
+		if exerciseConfig == nil {
+			return "", fmt.Errorf("exerciseConfig not initialize before use")
+		}
+		testFiles, err := exerciseConfig.GetTestFiles()
+		if err != nil {
+			return "", err
+		}
+		cmd = strings.ReplaceAll(cmd, "{{test_files}}", strings.Join(testFiles, " "))
+	}
+
+	return cmd, nil
+}
+
+var TestConfigurations = map[string]TestConfiguration{
```

Comment:
I've just done a quick double check to see which active tracks are missing.
This is the list of missing tracks:

- abap
- common-lisp
- cpp
- delphi
- fortran
- objective-c
- pharo-smalltalk
- plsql
- powershell
- r
- scheme
- unison
- vimscript

Could you double check these tracks? If they're not supported, it might be useful to add a comment somewhere noting that those tracks are not supported (and why)

Reply:
Sure, added in ad33864eff918818939e183f8d7151582638c43a! I've confirmed that most of the remaining tracks are N/A (tests are run in IDE, etc).

Reply:
For common-lisp, I used 
```sh
sbcl --load "${slug}-test" \
     --eval "(exit :code (if (${slug}-test:run-tests) 0 5))"
```
That assumes a particular distribution: there are several suggested on the track's intstallation page.

For fortran, I have this fish snippet
```fish
    if not test -d ./.exercism
        echo run me from the exercise root >&2
        return 1
    end
    if not test -d ./build
        mkdir -p build; and cd build
        test -d ./testlib; or cmake -G "Unix Makefiles" ..
        cd ..
    end
    cd build
    make; and ctest -V
    cd ..
```

Reply:
> R is just
> Rscript test_*.R

Could you update the configuration to use this command? I can confirm that this is also [how the test runner does it](https://github.com/exercism/r-test-runner/blob/main/bin/run.sh#L39).

Reply:
Ah, good call! I didn't even think about the test runners - I've been going off the public docs